main.rb

```diff
@@ -0,0 +1,19 @@
+def decode_morse_code(morse)
+  morse_to_ascii = {
+    '.-' => 'a', '-...' => 'b', '-.-.' => 'c', '-..' => 'd', '.' => 'e',
+    '..-.' => 'f', '--.' => 'g', '....' => 'h', '..' => 'i', '.---' => 'j',
+    '-.-' => 'k', '.-..' => 'l', '--' => 'm', '-.' => 'n', '---' => 'o',
+    '.--.' => 'p', '--.-' => 'q', '.-.' => 'r', '...' => 's', '-' => 't',
+    '..-' => 'u', '...-' => 'v', '.--' => 'w', '-..-' => 'x', '-.--' => 'y', '--..' => 'z',
+    '-----' => 0, '.----' => 1, '..---' => 2, '...--' => 3, '....-' => 4,
+    '.....' => 5, '-....' => 6, '--...' => 7, '---..' => 8, '----.' => 9, '$$$' => ' '
+  }
+  morse_arr = morse.gsub(/\s{3}/, ' $$$ ')
+  morse_arr = morse_arr.split(/\s/)
+  morse_arr.each { |n| print morse_to_ascii[n].upcase }
+  puts
+end
```

Comment:
- [ ] You have only one Morse decoding method, but from the project requirements, you should have three.

  - The first method for decoding one character.
  - The second for decoding a word using the first method.
  - The third is for decoding a sentence using the second method.

  Kindly implement that by splitting this method into 3 methods.

Reply:
Our previous code is the best, please read all the requirements.
We resolved it by reuse methods.